### PR TITLE
Feature/setup runtime feature flag

### PR DESCRIPTION
**Pull request by aliokan**
Add a runtime feature flag service. 

What was done : 
- add `FeaturesService`
- add `mu-search-agenda-items` flag as an example, will be used in https://github.com/lblod/frontend-burgernabije-besluitendatabank/pull/83

### CODE REVIEW

File: app/components/vote-overview.ts [hunk 1]
```diff
@@ -9,12 +9,6 @@ interface ArgsInterface {
   vote: VoteModel;
 }
 
-interface FormattedTableVote {
```
Comment (Windvis) [inline app/components/vote-overview.ts:12]: (nitpick, no need to change: this seems unrelated to this PR so ideally it would have been split into a separate PR.)

File: app/routes/agenda-items/index.ts [hunk 2]
```diff
@@ -15,4 +21,22 @@ export default class AgendaItemsIndexRoute extends Route {
...
+    if (this.features.isEnabled('mu-search-agenda-items')) {
+      console.log(
+        'mu-search-agenda-items feature is enabled, but no visible change yet'
+      );
+      // this.controllerName = 'agenda-items/mu-search';
+      // this.templateName = 'agenda-items/mu-search';
+    }
+
+    super.setupController(controller, model, transition);
+  }
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/routes/agenda-items/index.ts:41]: You can actually, it will be used in https://github.com/lblod/frontend-burgernabije-besluitendatabank/pull/83

File: README.md [hunk 3]
```diff
@@ -43,6 +43,51 @@ The project uses woodpecker to automate builds. The configuration files can be f
...
+let ENV = {
+    // Other configuration settings...
+    features: {
+      'new-feature': true,    // Enable the 'new-feature' by default
+      'beta-feature': false,  // Disable the 'beta-feature' by default
+    }
+  };
+```
+
+The configuration can be manually overridden by setting a variable in the browser console, feature name have to be prefixed with `feature-`:
```
[the remark this answers is not in the document]
Comment (aliokan) [reply README.md]: You right, a piece was missing. In the new version, you can override the configuration through query parameters. The overriding will be saved in a cookie, so it will persist across page reloads. The cookie can be cleared by adding `?clear-feature-overrides=true` to the URL.

File: config/environment.js [hunk 4]
```diff
@@ -20,6 +20,9 @@ module.exports = function (environment) {
       // Here you can pass flags/options to your application instance
       // when it is created
     },
+    features: {
+      'mu-search-agenda-items': false,
```
[the remark this answers is not in the document]
Comment (aliokan) [reply config/environment.js:24]: done https://github.com/lblod/frontend-burgernabije-besluitendatabank/pull/90/files#diff-f0ec527fe84535680fbd7291cd9f1b67feb487f2a99b963aabe646318adf5172R45-R50

File: app/services/features.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import config from 'frontend-burgernabije-besluitendatabank/config/environment';
+
+export default class FeaturesService extends Service {
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/services/features.ts:5]: I hesitated a lot, but in the end I chose a service. It adds a bit of constraint, but it improves the encapsulation and I have the feeling that it is more in the Ember philosophy. If it's really a blocker for you, I can turn it into a util.

File: app/routes/agenda-items/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Windvis) [reply app/routes/agenda-items/index.ts:41]: This just seemed like a change that would be part of the mu-search PR (after this one was merged). It added little value to this one, other than an example. I don't think we need the setupController for the real implementation either? Wouldn't we just check the feature flag in the AgendaItemsLoader?

Anyway, we can keep it if you want, but I think it will be removed again later 😄.

File: app/services/features.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Windvis) [reply app/services/features.ts:5]: Yea, I understand, but this implementation has some shortcomings, like I mentioned. There's a reason why I went with that function based implemenation before 😄.

Anyway, maybe we won't run into the shortcomings in this project (or there are workarounds to achieve the same goal). One thing that we do probably want is a helper version so we can check feature flags in templates as well. (The version is loket was both a helper and a util, so it covered both JS and .hbs files)

File: app/services/features.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    assert(
+      `The "${feature}" feature is not defined. Make sure the feature is defined in the "features" object in the config/environment.js file and that there are no typos in the name.`,
+      feature in this.#features
+    );
+
+    return this.#features[feature] ?? false;
+  }
+
+  #getConfigFeatures(): Record<string, boolean> {
+    const cookieFeatures: Record<string, boolean> = Object.fromEntries(
```
Comment (Windvis) [inline app/services/features.ts:46]:
```suggestion
    const configFeatures: Record<string, boolean> = Object.fromEntries(
```

File: app/services/features.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+    // save query params in cookie
+    if (Object.keys(queryFeatures).length > 0) {
+      this.#setCookieFeatures(queryFeatures);
+    }
+  }
+
+  setup(features: Record<string, boolean>) {
+    this.#features = { ...features };
+    console.log('Feature flags:', this.#features);
```
Comment (Windvis) [inline app/services/features.ts:33]:
```suggestion
```

File: app/services/features.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
+import Service from '@ember/service';
+import { assert } from '@ember/debug';
+import config from 'frontend-burgernabije-besluitendatabank/config/environment';
+
+export default class FeaturesService extends Service {
+  static PREFIX = 'feature-';
```
Comment (Windvis) [inline app/services/features.ts:6]: This doesn't seem to be used yet?

**Review by Windvis — APPROVED** [review]
Some small nitpicks. Works as described 👍.

File: app/services/features.ts (hunk 7 above)
Comment (aliokan) [reply app/services/features.ts:33]: I put it to simplify exchanges with future testers ;)

**Review by aliokan — COMMENTED** [review]
@Windvis :ping_pong:

**Review by Windvis — APPROVED** [review]
